Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions-details.tsx

```diff
@@ -117,6 +129,35 @@ const SubmissionsDetailsProvider = ({ children }: ISubmissionsDetailsProviderPro
         setCurrentProblemSubmissionResults(apiProblemResults as ISubmissionDetails[]);
     }, [ apiProblemResults ]);
 
+    const {
+        get: getProblemResultsByUserRequest,
+        data: apiProblemResultsByUser,
+    } = useHttp(getSubmissionResultsByProblemAndUserUrl as UrlType, submissionResultsByProblemAndUserUrlParams);
+
+    useEffect(
+        () => {
+            if (isNil(submissionResultsByProblemAndUserUrlParams)) {
+                return;
+            }
+
+            (async () => {
+                startLoading();
+                await getProblemResultsByUserRequest();
+                setSubmissionResultsByProblemAndUserUrlParams(null);
```

Comment:
The way I understand it is if the url params are not null, we are making the request using these params and after that we are setting them to null so that we do not make the request again. That being said, if we set them to null before the request, it still works, which is strange to me, so maybe there is something I am not getting here.